src/main/java/org/owasp/esapi/reference/validation/HTMLValidationRule.java

```diff
@@ -46,22 +46,29 @@
 	/** OWASP AntiSamy markup verification policy */
 	private static Policy antiSamyPolicy = null;
 	private static final Logger LOGGER = ESAPI.getLogger( "HTMLValidationRule" );
+	private static final String ANTISAMYPOLICY_FILENAME = "antisamy-esapi.xml";
 
 	static {
         InputStream resourceStream = null;
 		try {
-			resourceStream = ESAPI.securityConfiguration().getResourceStream("antisamy-esapi.xml");
+			resourceStream = ESAPI.securityConfiguration().getResourceStream(ANTISAMYPOLICY_FILENAME);
 		} catch (IOException e) {
-			throw new ConfigurationException("Couldn't find antisamy-esapi.xml", e);
-	            }
+			
+			LOGGER.info(Logger.EVENT_FAILURE, "Loading " + ANTISAMYPOLICY_FILENAME + " from classpaths");
+
+			resourceStream = ESAPI.securityConfiguration().getResourceStreamFromClasspath(ANTISAMYPOLICY_FILENAME);
+		}
         if (resourceStream != null) {
         	try {
 				antiSamyPolicy = Policy.getInstance(resourceStream);
 			} catch (PolicyException e) {
-				throw new ConfigurationException("Couldn't parse antisamy policy", e);
-		        }
-			}
+				throw new ConfigurationException("Couldn't parse " + ANTISAMYPOLICY_FILENAME, e);
+	        }
 		}
+        else {
+            throw new ConfigurationException("Couldn't find " + ANTISAMYPOLICY_FILENAME);
```

Comment:
I believe this exception should be thrown from the implementation in SecurityConfiguration.

Rather than making a null stream an undocumented valid return, I would suggest updating the documentation to note that if the resource cannot be located that a ConfigurationException will result. 

This reduces the amount of context that we require every client/caller to when using the API.

If you disagree with this assessment, then the documentation on the API should be updated to declare the functional behavior when the resource cannot be located IE:  `@return inputstream or null if resource cannot be located or loaded`.

Reply:
Worth noting that both Class and Classloader methods document that null is a valid return case.

https://docs.oracle.com/javase/7/docs/api/java/lang/ClassLoader.html#getSystemResourceAsStream(java.lang.String)
https://docs.oracle.com/javase/7/docs/api/java/lang/Class.html#getResourceAsStream(java.lang.String)

So contexually I would ask whether we expect that calls to the SecurityConfiguration will be for arbitrary files that may or may not exist, or files which are expected to be present for the environment in which ESAPI is running.

I tend toward the second option in my suggestion of moving the ConfigurationException into the SecurityConfiguration, but I also recognize that is just my opinion.

Reply:
Well, whether this should be a general method used to retrieve any resource from the class loaders is the $64,000 question now, isn't it? Based on the original Javadoc comments, it doesn't seem like that was the intent, even though this is what out would currently allow. So I'm absolutely with @jeremiahjstacey regarding the exception handling if we are to keep it in SecurityConfiguration. My first wish would be to make it a private static method in HTMLValidationRule and get it out of the SecurityConfiguration interface altogether unless there is some more general use case one could think of that we should be supporting with it. (I can think of none off the top of my head.) (Of course, moving it to there mentions duplicating even more code, but not that much. This would be a nice place to have something like C++'s 'friend' member functions.)

Reply:
No longer an issue since this change was moved to a new private method in HTMLValidationRule. Resolved.